test/e2e/containers/fake_datadog/app/api.py

```diff
@@ -69,14 +69,31 @@ def record_and_loads(filename: str, content_type: str, content_encoding: str, co
     return json.loads(content)
 
 
+def patch_data(data, patch_key, patch_leaf):
+    if isinstance(data, dict):
+        return {patch_key(k): patch_data(v, patch_key, patch_leaf) for k, v in iter(data.items())}
+    elif isinstance(data, list):
+        return [patch_data(i, patch_key, patch_leaf) for i in data]
+    else:
+        return patch_leaf(data)
+
+
+def fix_data(data):
+    return patch_data(
+        data,
+        patch_key=lambda x: x.translate(str.maketrans('.$', '\uff0e\uff04')),
```

Comment:
That's a bit sneaky :D, why not replacing with more common chars like `_`

Reply:
Well, I just found many StackOverflow or blog posts pointing to the [official MongoDB documentation](https://docs.mongodb.com/v2.6/faq/developers/#dollar-sign-operator-escaping) which says:
> In some cases, you may wish to build a BSON object with a user-provided key. In these situations, keys will need to substitute the reserved $ and . characters. Any character is sufficient, but consider using the Unicode full width equivalents: U+FF04 (i.e. “＄”) and U+FF0E (i.e. “．”).

So, doing this seemed to be the MongoDB idiomatic way of dealing with this issue.
But `_` would have definitively do the job as well.

Reply:
Fair enough, just put that in comment then